internal/buildkite/joblogs/process.go

```diff
@@ -0,0 +1,44 @@
+package joblogs
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/buildkite/go-buildkite/v4"
+	"github.com/buildkite/terminal-to-html/v3"
+	"github.com/huantt/plaintext-extractor"
+)
+
+var timeTagRegexp = regexp.MustCompile(`<time[^>]*>.*?</time>`)
+
+// Process accepts job logs from the Buildkite API and strips out formatting
+// to reduce the number of tokens sent to the LLM
+func Process(jobLog buildkite.JobLog) (string, error) {
+	screen, err := terminal.NewScreen()
+	if err != nil {
+		return "", fmt.Errorf("failed to create terminal screen: %w", err)
+	}
+
+	if _, err = screen.Write([]byte(jobLog.Content)); err != nil {
+		return "", fmt.Errorf("failed to write to terminal screen: %w", err)
+	}
+
+	html := screen.AsHTML()
+	output := strings.Builder{}
+	extractor := plaintext.NewHtmlExtractor()
+
+	for line := range strings.Lines(html) {
+		// remove timestamps to save a few more tokens
+		line = timeTagRegexp.ReplaceAllString(line, "")
+
+		plainText, err := extractor.PlainText(line)
+		if err != nil {
+			return "", fmt.Errorf("failed to extract plain text: %w", err)
+		}
```

Comment:
if plainText can is null, I would just return a empty string.

Also this avoids nil deference issues.

Reply:
Yeah, fair call. It should only be nil when err is not nil, but good to be defensive in case that changes. 

https://github.com/huantt/plaintext-extractor/blob/c8f302a12a28822f8008808e867d059975756e4d/parser.go#L47

Reply:
Yeah you can just add a safe ToString function like this.

https://github.com/aws/smithy-go/blob/main/ptr/from_ptr.go#L78-L87